### PR TITLE
OP-441 Keep CSI disabled for existing cluster, enable for new install

### PR DESCRIPTION
**Pull request by piyush-nimbalkar**
Signed-off-by: Piyush Nimbalkar <pnimbalkar@purestorage.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes # https://portworx.atlassian.net/browse/OPERATOR-441

**Special notes for your reviewer**:

### CODE REVIEW

**Review by adityadani — APPROVED** [review]

File: drivers/storage/portworx/portworx.go [hunk 1]
```diff
@@ -629,12 +629,15 @@ func setPortworxDefaults(toUpdate *corev1.StorageCluster) {
 		}
 	}
 
-	// Enable CSI flag by default
-	if _, ok := toUpdate.Spec.FeatureGates[string(pxutil.FeatureCSI)]; !ok {
-		if toUpdate.Spec.FeatureGates == nil {
-			toUpdate.Spec.FeatureGates = make(map[string]string)
+	if t.isK3s || toUpdate.Spec.Version == "" {
```
Comment (dsupure) [inline drivers/storage/portworx/portworx.go:632]: So k3s cannot use in-tree and only uses CSI?

File: drivers/storage/portworx/util/feature.go [hunk 2]
```diff
@@ -15,5 +15,5 @@ const (
 // IsEnabled checks if the feature is enabled in the given feature map
 func (feature Feature) IsEnabled(featureMap map[string]string) bool {
 	enabled, err := strconv.ParseBool(featureMap[string(feature)])
-	return err != nil || enabled
+	return err == nil && enabled
```
Comment (dsupure) [inline drivers/storage/portworx/util/feature.go:18]: Thanks for catching this minor bug.

**Review by dsupure — COMMENTED** [review]
Generally looking very good. LGTM!

File: drivers/storage/portworx/portworx.go (hunk 1 above)
Comment (piyush-nimbalkar) [reply drivers/storage/portworx/portworx.go:632]: Correct. In k3s in-tree plugins are disabled to make k8s lighter. CSI is the only way to interact with storage providers.

**Pull request merged**